app/src/main/java/com/asfoundation/wallet/my_wallets/name/NameDialogViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package com.asfoundation.wallet.my_wallets.name
+
+import androidx.lifecycle.SavedStateHandle
+import com.asfoundation.wallet.base.Async
+import com.asfoundation.wallet.base.BaseViewModel
+import com.asfoundation.wallet.base.SideEffect
+import com.asfoundation.wallet.base.ViewState
+import com.asfoundation.wallet.wallets.repository.WalletInfoRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+sealed class NameDialogSideEffect : SideEffect {
+  object NavigateBack : NameDialogSideEffect()
+}
+
+data class NameDialogState(
+  val walletAddress: String,
+  val walletNameAsync: Async<String> = Async.Uninitialized
+) : ViewState
+
+@HiltViewModel
+class NameDialogViewModel @Inject constructor(
+  savedStateHandle: SavedStateHandle,
+  private val walletInfoRepository: WalletInfoRepository,
+) :
+  BaseViewModel<NameDialogState, NameDialogSideEffect>(initialState(savedStateHandle)) {
+
+  companion object {
+    fun initialState(savedStateHandle: SavedStateHandle): NameDialogState = NameDialogState(
+      savedStateHandle.get<String>(NameDialogFragment.WALLET_ADDRESS_KEY)!!,
+      Async.Success(savedStateHandle.get<String>(NameDialogFragment.WALLET_NAME_KEY)!!),
+    )
+  }
+
+  fun setName(name: String) {
+    val retainValue = NameDialogState::walletNameAsync
+    state.walletNameAsync()?.also {
+      walletInfoRepository.updateWalletName(state.walletAddress, name)
```

Comment:
Shouldn't we use a useCase/interactor here? It's ok either way, but I feel it's better if we either always use useCases or never use them at all.

Reply:
Added use case for convenience